### ISSUE TITLE
Search bar not working

### ISSUE THREAD

**Issue by carlosgalvezp** [doxygen/doxygen#10417]
**Describe the bug**
We have recently bumped from 1.9.2 to 1.9.8. To our surprise, the search bar on the top-right corner no longer returns any result, regardless of what we type to search. The search functionality was working fine on 1.9.2. We used the built-in functionality in Doxygen to re-generate the Doxyfile file, which simply added a handful of new fields, but didn't change existing fields.

**Expected behavior**
The search bar provides search results.

**Screenshots**
![Screenshot from 2023-11-16 10-28-45](https://github.com/doxygen/doxygen/assets/1083865/04f6b92d-8815-4fac-b13f-60376a354597)

**Version**
1.9.8 (Ubuntu 20.04)

**Comment by albert-github** [comment]
I have not seen any problem regarding the search bar in this respect and without an example it is impossible to tell what is wrong, so:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by carlosgalvezp** [comment]
Here comes a self-contained minimal example. 
[doxygen_10417.zip](https://github.com/doxygen/doxygen/files/13411762/doxygen_10417.zip)

If I type `f` in the search bar, nothing happens.

If I remove the `|` on line 5, then search works - when I type `f`, I get a hit of the `Foo` struct.

This code was part of a larger ASCII drawing. Should it be put inside some verbatim block or something?

I still believe it's really bad that this breaks the search functionality for the entire project, without any warning or error.

**Comment by albert-github** [comment]
> This code was part of a larger ASCII drawing. Should it be put inside some verbatim block or something?

Yes I think this should be in a verbatim type of block as otherwise it will be regarded just as text and will be processed with all kind of (unwanted) effects.


> I still believe it's really bad that this breaks the search functionality for the entire project, without any warning or error.

It has to be checked what causes the breaking of the search functionality.
We have here not just the `|` but also the `#` which indicates here that a section starts. We can see this in the output:
![image](https://github.com/doxygen/doxygen/assets/5223533/2fd6bdcc-d682-4807-8325-5fe53bd9eb95)


but better when we look at the markdown intermediate output (`doxygen -d markdown`):
```
======== Markdown =========
---- input -------
The foo

Drawing:

 # |

Something else
---- output -----
The foo

Drawing:

@section autotoc_md0 |

Something else

=========
```

Here we see that 
```
# |
```
has been translated into 
```
@section autotoc_md0
```

**Comment by albert-github** [comment]
Did some further investigations.
Since doxygen 1.9.8 a.o. also section titles are searchable which results in the fact that when having:
```
# | A
```
and one type in the search bar `a` one sees the section title but one cannot search for the `|`.
When having though:
```
# _ A
```
one can search for `a` but also for `_` and this is due to the fact that e.g/ variable names can contain a `_` but (to the best of my knowledge) but not a `|` . 

When having just:
```
# _
```
one still can search for the `_` but when having:
```
# |
```
one cannot search for the `|` as the `|` is not a supported character.

**Comment by doxygen** [comment]
The problem was that a non-identifier character at the end of a title caused an empty string to be indexed, which then prevented other words to be found. Please verify if the referenced commit fixes the problem. Don't close the issue, this will be done automatically when the next version is released.

**Comment by albert-github** [comment]
The problem lies a bit in the fact that there is now the possibility that section titles are also searchable and thus can contain non identifier characters

When having:
```
/// The foo
///
///  # F G
///
///  # F _ G
///
struct Foo
{};

/// The doo
///
///  # D E
///
///  # D | E
///
struct Doo
{};
```
We can search for `F _ G` but we cannot search for `D | E`.

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/13464792/example.tar.gz)
